src/net_processing.cpp

```diff
@@ -2608,6 +2608,15 @@ bool ProcessMessage(CNode* pfrom, const std::string& msg_type, CDataStream& vRec
         }
 
         const CNetMsgMaker msg_maker(INIT_PROTO_VERSION);
+        // Signal ADDRv2 support (BIP155).
+        if (nSendVersion >= 70016) {
+            // BIP155 defines addrv2 and sendaddrv2 for all protocol versions, but some
+            // implementations reject messages they don't know. As a courtesy, don't send
+            // it to nodes with a version before 70016, as no software is known to support
+            // BIP155 that doesn't announce at least that protocol version number.
+            connman->PushMessage(pfrom, msg_maker.Make(NetMsgType::SENDADDRV2));
+        }
+
```

Comment:
isn't this before mnauth?

Reply:
yes, but it's before VERACK so it's fine

Reply:
so was the comment here wrong? https://github.com/dashpay/dash/pull/4874/files/fbacffe035ab51cdce82fa0c1a814ca1e28ba25d#diff-6875de769e90cec84d2e8a9c1b962cdbcda44d870d42e4215827e599e11e90e3L2717-L2720

L2717-2720

Reply:
oh, in that location it was after verack, so a problem, but here it's before both verack and mnauth.